make/compiler_flags

```diff
@@ -13,7 +13,7 @@ ifeq (default,$(origin CXX))
     CXX := clang++
   endif
   ifeq ($(OS),Linux)
-    CXX := g++
+    CXX := clang++
```

Comment:
Not sure how we should go about fixing this one, clang-tidy uses clang but I'd like to keep using g++ on linux

Reply:
Can we set a variable if clang-tidy is used and then set clang++ here?

Reply:
This comment would also apply here https://github.com/stan-dev/math/pull/1326#discussion_r318978191

Reply:
Yeah. That sounds right. 

We want to leave g++ as the default.

Reply:
Fixed! Turns out clang-tidy runs just fine with g++